Add ObservationEncodingsTest, SOS 1.0.0 OperationEncoderKey for coding-json GetObs

shane-axiom opened the pull request:
This PR:

* Adds an encoder key for SOS 1.0.0 to the `coding-json` `AbstractSosResponseEncoder`
* Factors out reusable methods from `ComplexObservationTest` to an extendable `AbstractObservationTest`
* Adds `ObservationEncodingsTest`, an integration test which inserts a simple sensor and observations and sends test `GetObservation` requests for various service versions and response formats.

CarstenHollmann commented on coding/json-common/src/main/java/org/n52/sos/encode/json/AbstractSosResponseEncoder.java:
```diff
@@ -49,8 +50,11 @@
+        super(type,
+                new OperationEncoderKey(SosConstants.SOS, Sos2Constants.SERVICEVERSION, operation,
+                        MediaTypes.APPLICATION_JSON),
+                new OperationEncoderKey(SosConstants.SOS, Sos1Constants.SERVICEVERSION, operation,
```
The JSON de-/encoding are only implemented for SOS 2.0 and O&M 2.0. The same for the JSON schemas.

So I think the JSON AbstractSosResponseEncoder should not support SOS 1.0.0.

shane-axiom commented:
@CarstenHollmann I see, I didn't realize the json encoding was bound to O&M 2.0. I removed the `OperationEncoderKey` for SOS 1.0.0 and adjusted the test. I also changed `AbstractServiceResponseWriter` to throw a `NoEncoderForKeyException` when an encoder can't be found (instead of wrapping it in a `RuntimeException`.
